### PR TITLE
add ability to configure ASH2 frame handler

### PR DESCRIPTION
**Pull request by mikomarrache**
Under certain configurations, the default hard coded ASH2 frame handler configuration is not appropriate. This PR simply provides the ability to client code to configure the frame handler.

The changes are backward compatible.

### CODE REVIEW

File: com.zsmartsystems.zigbee.dongle.ember/src/main/java/com/zsmartsystems/zigbee/dongle/ember/internal/ash/AshFrameHandler.java [hunk 1]
```diff
@@ -60,27 +60,85 @@
      */
     private final Logger logger = LoggerFactory.getLogger(AshFrameHandler.class);
 
-    /**
-     * The receive timeout settings - min/initial/max - defined in milliseconds
-     */
-    private final static int T_RX_ACK_MIN = 400;
```
Comment (cdjackson) [inline com.zsmartsystems.zigbee.dongle.ember/src/main/java/com/zsmartsystems/zigbee/dongle/ember/internal/ash/AshFrameHandler.java:66]: I would prefer to keep these definitions. These directly relate to UG101.

File: com.zsmartsystems.zigbee.dongle.ember/src/main/java/com/zsmartsystems/zigbee/dongle/ember/internal/ash/AshFrameHandler.java [hunk 2]
```diff
@@ -60,27 +60,85 @@
...
-    private final static int T_RX_ACK_MAX = 3200;
-    private int receiveTimeout = T_RX_ACK_INIT;
+    public static class AshFrameHandlerConfig {
 
-    private final static int T_CON_HOLDOFF = 1250;
-    private int connectTimeout = T_CON_HOLDOFF;
+        /**
+         * The receive timeout settings - min/initial/max - defined in milliseconds
+         */
+        private int ackReceiveMinTimeout = 400;
```
Comment (cdjackson) [inline com.zsmartsystems.zigbee.dongle.ember/src/main/java/com/zsmartsystems/zigbee/dongle/ember/internal/ash/AshFrameHandler.java:68]: Please refer to the constants - eg ```T_RX_ACK_MIN```

File: com.zsmartsystems.zigbee.dongle.ember/src/main/java/com/zsmartsystems/zigbee/dongle/ember/internal/ash/AshFrameHandler.java [hunk 3]
```diff
@@ -60,27 +60,85 @@
...
+         * Maximum number of consecutive timeouts allowed while waiting to receive an ACK
+         */
+        private int maxAckTimeouts = 4;
+
+        /**
+         * Maximum number of DATA frames we can transmit without an ACK
+         */
+        private int txWindow = 1;
+
+        public int getAckReceiveMinTimeout() {
```
Comment (cdjackson) [inline com.zsmartsystems.zigbee.dongle.ember/src/main/java/com/zsmartsystems/zigbee/dongle/ember/internal/ash/AshFrameHandler.java:84]: Please add Javadoc to the public methods.

File: com.zsmartsystems.zigbee.dongle.ember/src/main/java/com/zsmartsystems/zigbee/dongle/ember/ZigBeeDongleEzsp.java [hunk 4]
```diff
@@ -129,6 +129,8 @@
      */
     private EzspProtocolHandler frameHandler;
 
+    private AshFrameHandler.AshFrameHandlerConfig ashFrameHandlerConfig;
```
Comment (cdjackson) [inline com.zsmartsystems.zigbee.dongle.ember/src/main/java/com/zsmartsystems/zigbee/dongle/ember/ZigBeeDongleEzsp.java:132]: I think this class should not refer to ASH. How do we handle SPI or ASH3?

File: com.zsmartsystems.zigbee.dongle.ember/src/main/java/com/zsmartsystems/zigbee/dongle/ember/ZigBeeDongleEzsp.java [hunk 5]
```diff
@@ -267,6 +269,17 @@ public ZigBeeDongleEzsp(final ZigBeePort serialPort) {
         this(serialPort, EmberSerialProtocol.ASH2);
     }
 
+    /**
+     * Create a {@link ZigBeeDongleEzsp} with the default ASH2 frame handler.
+     * The provided config will be used to configure the ASH2 frame handler.
+     *
+     * @param serialPort the {@link ZigBeePort} to use for the connection
```
Comment (cdjackson) [inline com.zsmartsystems.zigbee.dongle.ember/src/main/java/com/zsmartsystems/zigbee/dongle/ember/ZigBeeDongleEzsp.java:276]: Javadoc missing

**Comment by mikomarrache** [comment]
We have managed to avoid tweaking the configuration. So I will close the PR in the meantime. Thanks

**Pull request closed without merging**